### DISCUSSION TITLE
Missing particles

### DISCUSSION THREAD

**Discussion by PRIYAGHEWADE** [NillionNetwork/discussions#46] [General]
Nada Program:

from nada_dsl import *

def nada_main():
    party1 = Party(name="Party1")
    my_array_1 = Array(SecretInteger(Input(name="my_array_1", party=party1)), size=1)
    my_int1 = SecretInteger(Input(name="my_int1", party=party1))

    @nada_fn
    def inc(a: SecretInteger, b: SecretInteger) -> SecretInteger:
        return a + b

    @nada_fn
    def mul(a: SecretInteger) -> SecretInteger:
        return a * my_int1

    new_array = my_array_1.map(mul).reduce(inc, my_int1)

    out = Output(new_array, "out", party1)

    return [out]

Nada Client:

Verified compute by retrieving output of compute and print on system,

Error:

nillion  | py_nillion_client.DealerError: computing: dealer initialization failed Booking service for compute failed: transaction failed: NotEnoughElementsAvailable()

After rebuilding same program without any changes we are getting,
nillion  | py_nillion_client.ComputeError: computing: compute unexpected error: missing particles 

 Client program:

    # Create Nillion Client
    client = py_nillion_client.NillionClient(
        nodekey,
        bootnodes,
        py_nillion_client.ConnectionMode.relay(),
        userkey,
        payments_config,
    )

    # SecretInteger as in my_program
    my_int1 = py_nillion_client.SecretInteger(1)
    
    
    combined_secrets = py_nillion_client.Secrets({"my_int1": my_int1})

    # We bind the storage of the secret to the circuit and the concrete party
    my_party_id = (
        client.party_id() if callable(getattr(client, "party_id")) else client.party_id
    )

    program_id = config["programs"]["basic"]
    bindings = py_nillion_client.ProgramBindings(program_id)
    bindings.add_input_party("Party1", my_party_id)
    bindings.add_output_party("Party1", my_party_id)

     
    print(f"Computing using program {program_id}") 

    # Store the first secret
    secreteStore = await client.store_secrets(
        cluster_id, bindings, combined_secrets, None
    )
    
    print(f"Stored secrets: {secreteStore}")
    # Do the computation, ensuring to pass an empty Secrets object if no additional secrets
    compute_id = await client.compute(
        cluster_id,
        bindings,
        [secreteStore],
        py_nillion_client.Secrets({}),  # Pass an empty Secrets object
        py_nillion_client.PublicVariables({}),
    )

    print(f"Computation sent to the network, compute_id = {compute_id}")  
    while True:
        event = await client.next_compute_event()      

        if isinstance(event, py_nillion_client.ComputeFinishedEvent):
            compute_output = event.result.value
            print(
                f"Received computation result for {event.uuid}, value = {event.result.value}"
            )
            break

    print("Retrieved Result for Circuit : {compute_output}")  

asyncio.run(main())

**Comment by wwwehr** [comment]
hey @PRIYAGHEWADE 

the only thing I'm spotting here is that the Arrays structures are experimental and are being considered for removal in a future versions. we suggest that you use python lists. 

See note here: https://github.com/orgs/NillionNetwork/discussions/40

**Comment by Davetbutler** [comment]
Hey @PRIYAGHEWADE 

Dave from the product team here. I was just wondering how your journey with Nillion was going, have you been able to build anything else yet?

[Maybe you are interested in our new builder bounties where you can win lots of prizes](https://github.com/orgs/NillionNetwork/projects/3)
